### PR TITLE
Viewport

### PR DESCRIPTION
**Pull request by biharck**
Adding the ability to use hotkeys with:
```
Default Tool
Zoom
Pan
Angle measurement
Scroll stack
Length measurement
Flip Horizontally
Flip Vertically
Rotate Right
Rotate Left
Invert
Zoom In
Zoom Out
Zoom to Fit
Reset
Clear Tools
Scroll Down
Scroll Up
Scroll to the Last Image
Scroll to First Image
```

### CODE REVIEW

File: src/CornerstoneViewport/CornerstoneViewport.js [hunk 1]
```diff
@@ -713,6 +724,96 @@ class CornerstoneViewport extends Component {
...
+      if (viewport.resetViewport) {
+        viewport = defaultViewport;
+      }
+
+      if (viewport.clearTools) {
+        const toolStateManager =
+          cornerstoneTools.globalImageIdSpecificToolStateManager;
+        toolStateManager.clear(this.element);
+        cornerstone.updateImage(this.element);
+      }
```
Comment (dannyrb) [inline src/CornerstoneViewport/CornerstoneViewport.js]: is `clearTools` a bool? To trigger this and other bool props again, do you need to quickly flip them after this call?

**Comment by dannyrb** [comment]
@biharck, is the idea here that something external updates the `viewport` prop for the `CornerstoneViewport`. When a change to the props is detected, you update the `CornerstoneViewport`'s `viewport` to reflect the changes?

If so, I might suggest the approach we discussed in Slack:

```
const mapStateToProps = state => {
  const { viewportSpecificData, activeViewportIndex } = state.viewports;
  const { dom } = viewportSpecificData[activeViewportIndex] || {};
  ...
}
```

Grab the `enabledElement` (dom) reference for whichever viewport you want to apply your command/action to, then apply the change to the viewport using cornerstone.

I'm not sure it's worth the round-trip complexity of having a reactive `viewport`, and this allows you to keep your code in a single library.

**Comment by dannyrb** [comment]
For example, this would turn your scrolldown logic into:

```
        let stack = this.state.stack;
        stack.currentImageIdIndex = Math.max(0, --stack.currentImageIdIndex);
        scrollToIndex(dom, stack.currentImageIdIndex);
        this.setState({ stack });
```

File: src/CornerstoneViewport/CornerstoneViewport.js (hunk 1 above)
Comment (biharck) [reply src/CornerstoneViewport/CornerstoneViewport.js]: You mean every time the user clicks? If so, then yes. 

https://github.com/OHIF/ohif-core/blob/viewport/src/classes/HotkeysUtil.js#L135

**Pull request merged**